### PR TITLE
[wordpress__blocks] Updated typings for version 11.0

### PR DESCRIPTION
**Pull request by marekdedic**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/WordPress/gutenberg/blob/trunk/packages/blocks/CHANGELOG.md
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by marekdedic** [comment]
Depends on #59511

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## wordpress__blocks ([<kbd>unpkg</kbd>](https://unpkg.com/browse/wordpress__blocks@latest/))
was missing the following properties:
1. store
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d46e8aa38d5aed806e571150d5ac3eb9fe37cb87
</p>

**Comment by marekdedic** [comment]
Seems like https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59550 fixed the CI issues :tada:

File: types/wordpress__blocks/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Type definitions for @wordpress/blocks 9.1
+// Type definitions for @wordpress/blocks 11.0
```
Comment (sirreal) [inline types/wordpress__blocks/index.d.ts:1]: [👍 Looks good for the upgraded dep](https://unpkg.com/browse/@wordpress/blocks@11.0.0/package.json)

File: types/wordpress__blocks/api/registration.d.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export function getBlockSupport<T>(
  * Returns a registered block type.
  */
 // tslint:disable:no-unnecessary-generics
-export function getBlockType<T extends Record<string, any> = any>(name: string | undefined): Block<T> | undefined;
+export function getBlockType<T extends Record<string, any> = {}>(name: string | undefined): Block<T> | undefined;
```
Comment (sirreal) [inline types/wordpress__blocks/api/registration.d.ts:23]: Was this change necessary?

**Review by sirreal — COMMENTED** [review]
The version bump looks good, will you provided details about the other change?

File: types/wordpress__blocks/api/registration.d.ts (hunk 2 above)
Comment (marekdedic) [reply types/wordpress__blocks/api/registration.d.ts:23]: Oh, it wasn't, it's just a leftover after merging #59550

**Comment by marekdedic** [comment]
Ad failing test: I think it's unrelated to the change and maybe just a fluke?

**Review by sirreal — APPROVED** [review]

**Review by andrewbranch — APPROVED** [review]

**Comment by marekdedic** [comment]
Ready to merge
